File: python/34b_1722A_SpellCheck.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        s = input().strip()
        if n != 5:
            ans.append("NO")
            continue
        flag = 0
        check = {"T" : 1,
             "i" : 1,
             "m" : 1,
             "u" : 1,
             "r" : 1}
        for i in s:
            if i in check:
                if check[i] == 0:
                    flag = 1
                    break
                else:
                    check[i] -= 1
            else:
                flag = 1
                break
        if flag == 0:
            ans.append("YES")
        else:
            ans.append("NO")
    for i in ans:
        print(i)

if __name__ =="__main__":
    main()

File: cpp/34b_1722A_SpellCheck.cpp
#include<bits/stdc++.h>

int main(){
    int t, n, flag;
    std::string s;
    std::vector<std::string> ans;
    std::unordered_map<char, int> check;
    check['T'] = 1;
    check['i'] = 1;
    check['m'] = 1;
    check['u'] = 1;
    check['r'] = 1;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>n;
        std::cin>>s;
        std::unordered_map<char, int> hmap;
        for (auto i : s){
            hmap[i] ++;
        }
        flag = 0;
        for(auto i = check.begin(); i != check.end(); i++){
            if(i->second != hmap[i->first]){
                flag = 1;
            }
        }
        if(check.size() != hmap.size()){
            flag = 1;
        }
        if(flag == 0){
            ans.push_back("YES");
        }
        else{
            ans.push_back("NO");
        }
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}